components/collapsed_reply_threads_tour/collapsed_reply_threads_modal/collapsed_reply_threads_modal.tsx

```diff
@@ -0,0 +1,113 @@
+
+    const onHide = useCallback((skipTour: boolean) => {
+        dispatch(closeModal(ModalIdentifiers.COLLAPSED_REPLY_THREADS_MODAL));
+        if (skipTour) {
+            const preferences = [{
+                user_id: currentUserId,
+                category: Preferences.CRT_TUTORIAL_TRIGGERED,
+                name: currentUserId,
+                value: (Constants.CrtTutorialTriggerSteps.FINISHED).toString(),
+            }];
+            dispatch(savePreferences(currentUserId, preferences));
+        }
+    }, []);
+
+    const onNext = useCallback(() => {
+        const preferences = [{
+            user_id: currentUserId,
+            category: Preferences.CRT_TUTORIAL_TRIGGERED,
+            name: currentUserId,
+            value: (Constants.CrtTutorialTriggerSteps.STARTED).toString(),
+        }];
+        dispatch(savePreferences(currentUserId, preferences));
+        onHide(false);
+    }, []);
+
+    return (
+        <GenericModal
+            className='CollapsedReplyThreadsModal productNotices'
+            id={ModalIdentifiers.COLLAPSED_REPLY_THREADS_MODAL}
+            enforceFocus={false}
+            onExited={props.onExited}
+            handleConfirm={onNext}
+            autoCloseOnConfirmButton={true}
+            handleCancel={() => onHide(true)}
+            modalHeaderText={(
+                <FormattedMessage
+                    id='collapsed_reply_threads_modal.title'
+                    defaultMessage={'A new way to view and follow threads'}
+                />
+            )}
+            confirmButtonText={(
+                <>
+                    <FormattedMessage
+                        id={'collapsed_reply_threads_modal.take_the_tour'}
+                        defaultMessage='Take the Tour'
+                    />
+                    <NextIcon/>
+                </>
+            )}
+            cancelButtonText={
+                <>
+                    <FormattedMessage
+                        id={'collapsed_reply_threads_modal.skip_tour'}
+                        defaultMessage='Skip Tour'
+                    />
+                </>
+            }
+        >
+            <div>
+                <p className='productNotices__helpText'>
+                    <FormattedMarkdownMessage
+                        id={'collapsed_reply_threads_modal.description'}
+                        defaultMessage={'Threads have been revamped to help you create organized conversation around specific messages. Now, channels will appear less cluttered as replies are collapsed under the original message, and all the conversations you\'re following are available in your **Threads** view. Take the tour to see what\'s new.'}
+                    />
+                </p>
+                <img
+                    src={crtBetaImg}
+                    className='CollapsedReplyThreadsModal__img'
+                />
+            </div>
+        </GenericModal>
+    );
+}
+
+export default memo(CollapsedReplyThreadsModal);
```

Comment:
I am currently questioning if we need to memoize the component at all.

If I am getting this right the component will only render once and has no interactional elements that stay mounted once interacted with. It either closes the onboarding flow, or takes the user to the next step, which will render a new component.

Correct me if I am wrong, but in that case memoizing makes the component more complex and less performant (since it uses caching mechanisms behind the scenes).